his_checker.py

```diff
@@ -9,9 +9,44 @@
 
 import sys
 import argparse
+import os
+
+COMMENTS_THRESHOLD = 0.2
+
+def process_comments(files):
+
+    """Process comment density within a file"""
+
+    metric_fail = 0
+    statements_tool = "pmccabe -c "
+    pmccabe_stat_index = 2
+    total_statements = 0
+    comments_tool = "pygount --format=cloc-xml "
+    comment_index = 21
+    comment_density = 0
+    license_comment_size = 4
+    print("Checking the file comment density")
+    for file in files.files:
+        sline = os.popen(statements_tool + str(file)).read().split('\n')
+        for fields in [line.split('\t') for line in sline[:-1]]:
+            statements = int(fields[pmccabe_stat_index])
+            total_statements = total_statements + statements
+        lines = os.popen(comments_tool + str(file)).read().split(' ')
+        if lines[comment_index].startswith("comment"):
+            comments = int(lines[comment_index].split("=")[1].replace('"', ''))
+            comments = comments - license_comment_size
+            comment_density = comments/total_statements
```

Comment:
This is going to throw an error on a division by zero if the parsed file has no statements - for example an header file.